src/custom/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,318 @@
+}: ShareModalProps): JSX.Element => {
+  const theme = useTheme();
+  const [openMenu, setMenu] = useState<boolean>(false);
+  const [selectedOption, setOption] = useState<string | undefined>(selectedResource?.visibility);
+  const [shareUserData, setShareUserData] = useState<User[]>([]);
+
+  const handleDelete = (email: string) => {
+    setShareUserData((prevData) => prevData.filter((user) => user.email !== email));
+  };
+
+  const handleOptionClick = (event: SelectChangeEvent<unknown>) => {
+    const value = event.target.value as string;
+    setOption(value);
+  };
+
+  const handleMenuClose = () => setMenu(false);
+
+  /**
+   * Copy design link in clipboard
+   */
+  const handleCopy = () => {
+    navigator.clipboard.writeText(resourceURL);
+  };
+
+  const isShareDisabled = () => {
+    const existingAccessIds = shareUserData.map((user) => user.id);
+    const ownerDataId = ownerData?.id;
+
+    if (ownerDataId) {
+      existingAccessIds.push(ownerDataId);
+    }
+
+    const hasMismatchedUsers = !shareUserData.every((user) => existingAccessIds.includes(user.id));
+
+    return (
+      shareUserData.length === existingAccessIds.length &&
+      !hasMismatchedUsers &&
+      (selectedOption === selectedResource?.visibility ||
+        shareUserData.length !== existingAccessIds.length)
+    );
+  };
+
+  useEffect(() => {
+    const fetchActors = async () => {
+      const actors = await fetchAccessActors();
+      setShareUserData(actors);
+    };
+    fetchActors();
+  }, [fetchAccessActors]);
+
+  useEffect(() => {
+    if (selectedResource) {
+      setOption(selectedResource?.visibility);
+    }
+  }, [selectedResource]);
+
+  return (
+    <div style={{ marginBottom: '1rem' }}>
+      <Modal
+        open={true}
+        closeModal={handleShareModalClose}
+        title={`Share ${dataName} "${selectedResource?.name}"`}
+      >
+        <ModalBody>
+          <UserSearchField
+            setUsersData={setShareUserData}
+            usersData={shareUserData}
+            label="Search Users"
+            customUsersList={
+              <AccessList
+                accessList={shareUserData}
+                ownerData={ownerData}
+                handleDelete={handleDelete}
+                hostURL={hostURL}
+              />
+            }
+            fetchSuggestions={fetchSuggestions}
+          />
+        </ModalBody>
+        <ModalBody>
+          <CustomListItemText>
+            <Typography variant="h6">General Access</Typography>
+          </CustomListItemText>
+          <CustomDialogContentText>
+            <FormControlWrapper size="small">
+              <div style={{ display: 'flex', justifyContent: 'start', alignItems: 'center' }}>
+                <VisibilityIconWrapper>
+                  {selectedOption === SHARE_MODE.PUBLIC ? (
+                    <PublicIcon
+                      width={24}
+                      height={24}
+                      stroke={theme.palette.mode === 'dark' ? '#fff' : 'black'}
```

Comment:
Theme color

Reply:
Will add

Reply:
oh you already mentioned here, Didn't saw